packages/grid/_modules_/grid/components/cell/GridEditDateCell.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import InputBase, { InputBaseProps } from '@material-ui/core/InputBase';
+import { GRID_CELL_EDIT_PROPS_CHANGE } from '../../constants/eventsConstants';
+import { GridCellParams } from '../../models/params/gridCellParams';
+
+export function GridEditDateCell(props: GridCellParams & InputBaseProps) {
+  const {
+    api,
+    cellMode,
+    colDef,
+    field,
+    formattedValue,
+    getValue,
+    hasFocus,
+    id,
+    isEditable,
+    row,
+    value,
+    ...other
+  } = props;
+
+  const [valueState, setValueState] = React.useState(value);
+
+  const handleChange = React.useCallback(
+    (event) => {
+      const newValue = event.target.value;
+      setValueState(newValue);
+
+      if (newValue === '') {
+        const editProps = { value: null };
+        // TODO use api.setEditCellProps
+        api.publishEvent(GRID_CELL_EDIT_PROPS_CHANGE, { id, field, props: editProps }, event);
+        return;
+      }
+
+      // TODO move to value parser
```

Comment:
I'm not sure it would make a difference anymore since we have custom logic in the filtering for dates.